docs/topics/advent-of-code.md

```diff
@@ -0,0 +1,74 @@
+[//]: # (title: Advent of Code puzzles in idiomatic Kotlin)
+
+[Advent of Code](https://adventofcode.com/) is an annual series of Christmas-themed puzzles that follow an Advent calendar
+when each puzzle is released daily. With the kind permission of [Eric Wastl](http://was.tl/), creator of Advent of Code,
+we demonstrate how to solve these puzzles using the idiomatic Kotlin style.
+
+> You can find all solutions for Advent of Code 2020 puzzles in our [GitHub repository](https://github.com/kotlin-hands-on/advent-of-code-2020/).
+>
+
+## Day 1. Report repair
+
+In this puzzle, you'll explore input handling, iteration over the list, different ways of building a map, and use
+the `let` function to simplify your code.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/1)
+* Check out the solution by Svetlana Isakova in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/07/advent-of-code-in-idiomatic-kotlin/)
+or watch the video:
+
+<video width="560" height="315" href="o4emra1xm88" title="Kotlin Tutorial: Advent of Code Puzzles, Day 1"/>
+
+## Day 2. Password philosophy
+
+In this puzzle, you'll explore string utility functions, regular expressions, operations on collections,
+and how the `let` function can be helpful to transform your expressions.
+
+* Read the puzzle description on [Advent of Code](https://adventofcode.com/2020/day/2)
+* Check out the solution by Svetlana Isakova in [Kotlin Blog](https://blog.jetbrains.com/kotlin/2021/07/advent-of-code-in-idiomatic-kotlin-day2/)
+or watch the video:
+
+<video width="560" height="315" href="MyvJ7G6aErQ" title="Kotlin Tutorial: Advent of Code Puzzles, Day 2"/>
+
+## Day 3. Toboggan trajectory
+
+In this puzzle, you'll compare imperative and more functional code styles, work with pairs and the `reduce` function,
```

Comment:
If it is not a scope function — use brackets after the function name (see our guidelines for the context).